File: ios/TheHoccSinhDienTu/TheHoccSinhDienTu/Models/Account.swift
//
//  Account.swift
//  TheHoccSinhDienTu
//
//  Created by Tondz on 5/31/22.
//  Copyright © 2022 Tondz. All rights reserved.
//

import Foundation
class Account{
    var account:String;
    var idHocSinh: Int;
    var idGiaoVien: Int;
    var password: String;
    var history: String;
    var idAccount: Int;
    init()  {
        account = "";
        idHocSinh = -1;
        idGiaoVien = -1;
        password = "";
        history = "";
        idAccount = -1;
    }
    
}

File: Backend/Models/Account.js
const { Sequelize, Model, DataTypes } = require("sequelize");
const { sequelize } = require('../Dao/Connection')
// const { Sequelize } = require('sequelize');
// Option 3: Passing parameters separately (other dialects)

// console.log(sequelize)
class Acount extends Model { };
Acount.init({
  account: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  idHocSinh: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  idGiaoVien: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  password: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  history: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  created_by_user_id: {
    type: DataTypes.INTEGER,
    allowNull: true
  },
  created_date_time: {
    type: DataTypes.DATE,
    allowNull: true
  },
  active_flag: {
    type: DataTypes.INTEGER,
    allowNull: true
  },
  updated_date_time: {
    type: DataTypes.DATE,
    allowNull: true
  },
  updated_by_user_id: {
    type: DataTypes.INTEGER,
    allowNull: true
  },
  idAcount: {
    type: DataTypes.INTEGER,
    allowNull: false,
    primaryKey: true,
    autoIncrement: true
  }
}, {
  sequelize, modelName: 'Acount'
});

module.exports.Acount = Acount;